### ISSUE TITLE
Expose the coordinator zome hashes of an installed app

### ISSUE THREAD

**Issue by mattyg** [holochain/holochain#2820]
The conductor knows which coordinator zomes are installed for each cell, but there is no way to ask it for their hashes. Two parts of the coordinator upgrades design need them: the remote call guard (#4912), where a caller has to learn the hash it wants to pin to, and capability grants (#4911), where a grant is bound to a coordinator hash and an operator needs to see which grants are still live.

Design: [`docs/design/coordinator_upgrades.md`](https://github.com/holochain/holochain/blob/develop/docs/design/coordinator_upgrades.md)

The coordinator hash needs to be stored in the install app data model rather than computed on demand. It could then be included in the `AppInfo` response rather than adding a new type and the lookup would be natural.

## Acceptance criteria

- The coordinator zome names and hashes of an installed app are readable over the app API.
- The values reflect what is installed now, so they change after an app update.

## Scope change from the original request

This issue originally asked for integrity, coordinator and UI hashes, read both from installed apps and from bundles on disk, so that a runtime could work out whether an update was coordinator-only or needed a forked install. That decision no longer belongs to the runtime. `update_app` (#4570) takes the desired state and either applies it or reports that the role needs a migration, so there is nothing for a runtime to diff beforehand. UI hashes were always the runtime's concern rather than Holochain's, and inspecting bundles on disk belongs in `hc` rather than the conductor. What remains is the narrow read above.

**Comment by nphias** [comment]
yes i was going to raise a similar feature request.  having an app signature of some kind ... a hash of all the authors work,  coordinator zomes, integrity, other crates etc..  so we know the app we are using is the official one etc.

**Comment by mjbrisebois** [comment]
There is something like this built into the redesigned devhub (see [`token structs`](https://github.com/holochain/devhub-dnas/blob/redesign-2.0/dnas/apphub/types/src/lib.rs#L166-L177)), but it would be great to have more insight into the installed app states.

**Comment by ThetaSinner** [comment]
- UI hashes are a problem for the runtime, not for Holochain, because UIs aren't stored on Holochain
- Coordinator hashes may not be relevant because with coordinator updates, you will be able to do a coordinator only update and it will just apply the latest state from your bundle. So hashes would really just be informational
- DNA migrations are a separate problem again, but we can get DNA hashes from Holochain, just not from bundles

We should discuss this some time around when we do coordinator updates and DNA migrations in the core usability epic
